zerver/lib/actions.py

```diff
@@ -5512,17 +5518,16 @@ def do_update_user_presence(
     # times per minute with multiple connected browser windows.
     # We also need to be careful not to wrongly "update" the timestamp if we actually already
     # have newer presence than the reported log_time.
-    if (
-        not created
-        and log_time - presence.last_active_time > datetime.timedelta(seconds=55)
```

Comment:
I think it was a bug to be looking at timedelta from `last_active_time` here. It would have meant updating the UserPresence row each time we get a new IDLE ping, if last_active_time was more than 55s ago.